File: Examples/VisualBasic/Thunderbird/Data/GetMarkerInformation.vb
Imports System.IO
Imports Aspose.Email.Formats.Mbox
Imports Aspose.Email.Mail

Public Class GetMarkerInformation
    Private Shared Sub Run()
        'ExStart: GetMarkerInformation
        '<summary>
        'This exmaple shows how to get marker information while reading or writing a message to Mbox file.
        'Introduced in Aspose.Email for .NET 6.4.0
        '</summary>
        Using stream As New FileStream("inbox.dat", FileMode.Open, FileAccess.Read)
            Using reader As New MboxrdStorageReader(stream, False)
                Dim fromMarker As String = Nothing
                Dim msg As MailMessage = reader.ReadNextMessage(fromMarker)
                'While (InlineAssignHelper(msg, reader.ReadNextMessage(fromMarker))) IsNot Nothing
                Do While Not msg Is Nothing

                    Console.WriteLine(fromMarker)

                    msg.Dispose()
                    msg = reader.ReadNextMessage(fromMarker)
                Loop
        End Using
        End Using
        Using writeStream As New FileStream("inbox.dat", FileMode.Create, FileAccess.Write)
            Using writer As New MboxrdStorageWriter(writeStream, False)
                Dim fromMarker As String = Nothing
                Dim msg As MailMessage = MailMessage.Load("input.eml")
                writer.WriteMessage(msg, fromMarker)

                Console.WriteLine(fromMarker)
            End Using
        End Using
        'ExEnd: GetMarkerInformaiton
    End Sub
End Class

File: Examples/CSharp/Thunderbird/GetMarkerInformation.cs
using Aspose.Email.Storage.Mbox;
using System;
using System.IO;

namespace Aspose.Email.Examples.CSharp.Email
{
    class GetMarkerInformation
    {
        public static void Run()
        {
            // ExStart: GetMarkerInformation
            // The path to the File directory.
            string dataDir = RunExamples.GetDataDir_Thunderbird();

            using (FileStream stream = new FileStream(dataDir + "ExampleMbox.mbox", FileMode.Open, FileAccess.Read))
            using (MboxrdStorageReader reader = new MboxrdStorageReader(stream, false))
            {
                MailMessage msg;
                string fromMarker = null;
                while ((msg = reader.ReadNextMessage(out fromMarker)) != null)
                {
                    Console.WriteLine(fromMarker);

                    msg.Dispose();
                }
            }

            using (FileStream writeStream = new FileStream(dataDir + "ExampleMbox.mbox", FileMode.Create, FileAccess.Write))
            using (MboxrdStorageWriter writer = new MboxrdStorageWriter(writeStream, false))
            {
                string fromMarker = null;
                MailMessage msg = MailMessage.Load(dataDir + "EmailWithAttandEmbedded.eml");
                writer.WriteMessage(msg, out fromMarker);

                Console.WriteLine(fromMarker);
            }
            // ExEnd: GetMarkerInformation
        }
    }
}
